File: Dp11727.c
#include <stdio.h>

int main(void){
    int memory[1001] = {};
    int N;
    scanf("%d", &N);

    memory[1] = 1;
    memory[2] = 3;
    int i=0;
    for(i=3; i<=N; i++){
        memory[i] = (memory[i-1] + memory[i-2]*2) % 10007;
    }

    printf("%d\n", memory[N]);
    return 0;
}

File: Dp11727.java
import java.util.*;

public class Dp11727{
    public static void main(String[] args){
        Scanner sc = new Scanner(System.in);
        final int N = sc.nextInt();
        int[] memory = new int[1001];
        memory[1] = 1;
        memory[2] = 3;

        for(int i=3; i<=N; i++){
            memory[i] = (memory[i-1] + (memory[i-2] * 2)) % 10007;
        }

        System.out.println(memory[N]);
    }
}